src/store/modules/util/actions.ts

```diff
@@ -585,9 +585,134 @@ const actions: ActionTree<UtilState, RootState> = {
+          "enumTypeId": "PROD_STR_STNG",
+          "description": "Identification preference to be used for scanning items.",
+          "enumName": "Barcode Identification Preference",
+          "enumCode": "BARCODE_IDEN_PREF"
+        })
+
+        if(hasError(resp)) {
+          throw resp.data;
+        }
+      }
+
+      const params = {
+        fromDate,
+        "productStoreId": ecomStore.productStoreId,
+        "settingTypeEnumId": "BARCODE_IDEN_PREF",
+        "settingValue": "internalName"
+      }  
+
+      await UtilService.createBarcodeIdentificationPref(params) as any
+    } catch(err) {
+      console.error(err)
+    }
+
+    // not checking for resp success and fail case as every time we need to update the state with the
+    // default value when creating a scan setting
+    commit(types.UTIL_BARCODE_IDENTIFICATION_PREF_UPDATED, "internalName")
+    return fromDate;
+  },
+
+  async setBarcodeIdentificationPref({ commit, dispatch, state }, value) {
+    let prefValue = state.barcodeIdentificationPref
+    const eComStoreId = store.getters['user/getCurrentEComStore'].productStoreId;
+
+    // when selecting none as ecom store, not updating the pref as it's not possible to save pref with empty productStoreId
+    if(!eComStoreId) {
+      showToast(translate("Unable to update barcode identification preference since no product store config found."))
+      commit(types.UTIL_BARCODE_IDENTIFICATION_PREF_UPDATED, prefValue)
+      return;
+    }
+
+    let fromDate;
+
+    try {
+      const resp = await UtilService.getProductStoreSetting({
+        "inputFields": {
+          "productStoreId": eComStoreId,
+          "settingTypeEnumId": "BARCODE_IDEN_PREF"
+        },
+        "filterByDate": 'Y',
+        "entityName": "ProductStoreSetting",
+        "fieldList": ["fromDate"],
+        "viewSize": 1
+      }) as any
+      if(!hasError(resp)) {
+        fromDate = resp.data.docs[0]?.fromDate
+      }
+    } catch(err) {
+      console.error(err)
+    }
+
+    if(!fromDate) {
+      fromDate = await dispatch("createBarcodeIdentificationPref");
+    }
+
+    const params = {
+      "fromDate": fromDate,
+      "productStoreId": eComStoreId,
+      "settingTypeEnumId": "BARCODE_IDEN_PREF",
+      "settingValue": value
+    }
+
+    try {
+      const resp = await UtilService.updateBarcodeIdentificationPref(params) as any
+
+      if((!hasError(resp))) {
+        showToast(translate("Barcode identification preference updated successfully."))
+        prefValue = value
+      } else {
+        throw resp.data;
+      }
+    } catch(err) {
+      showToast(translate("Failed to update barcode identification preference."))
+      console.error(err)
+    }
+    commit(types.UTIL_BARCODE_IDENTIFICATION_PREF_UPDATED, prefValue)
+  },
   
   async updateForceScanStatus({ commit }, payload) { 
     commit(types.UTIL_FORCE_SCAN_STATUS_UPDATED, payload)
+  },
+
+  async updateBarcodeIdentificationPref({ commit }, payload) { 
```

Comment:
This action is not used anywhere, we should use this method wherever we are updating the state for barcode identifier, also we need to clear the barcode identifier value of logout.